core/unit_test/TestMathematicalSpecialFunctions.hpp

```diff
@@ -694,51 +694,61 @@ struct TestComplexBesselJ0Y0Function {
 
   KOKKOS_INLINE_FUNCTION
   void operator()(const int& i) const {
-    d_cbj0(i) = Kokkos::Experimental::cyl_bessel_j0<Kokkos::complex<double>,
-                                                    double, int>(d_z(i));
-    d_cby0(i) = Kokkos::Experimental::cyl_bessel_y0<Kokkos::complex<double>,
-                                                    double, int>(d_z(i));
+    d_cbj0(i) =
+        Kokkos::Experimental::Impl::cyl_bessel_j0<Kokkos::complex<double>,
+                                                  double, int>(d_z(i));
+    d_cby0(i) =
+        Kokkos::Experimental::Impl::cyl_bessel_y0<Kokkos::complex<double>,
+                                                  double, int>(d_z(i));
   }
 
   KOKKOS_INLINE_FUNCTION
   void operator()(const TestLargeArgTag&, const int& /*i*/) const {
     d_cbj0_large(0) =
-        Kokkos::Experimental::cyl_bessel_j0<Kokkos::complex<double>, double,
-                                            int>(d_z_large(0));
+        Kokkos::Experimental::Impl::cyl_bessel_j0<Kokkos::complex<double>,
```

Comment:
Why did you move `cyl_bessel_*[0|1]` to `Impl`? If we really want to remove them from the public interface, we should rather deprecate them (in the public interface).

Reply:
I moved it there so user will not call  `cyl_bessel_*[0|1]`since native functions can exist for cuda or std, and it creates more uniform interface 
if exist better solution to hide this functions please tell me